### ISSUE TITLE
Mysterious math function using clobbered variable?

### ISSUE THREAD

**Issue by alerque** [sile-typesetter/sile#422]
I can't seem to make a simple function behave here and am pulling my hair out. I have two bits, either of which work on their own, but that _can_ work together but also _can_ fail. Here's the code:

```Lua
SILE.registerCommand("footnote:counter", function(options, content)
  SILE.call("noindent")
  SILE.typesetter:pushGlue({ width = SILE.length.new({ length = 0 }) - SILE.length.parse(options.indent) })
  SILE.call("rebox", { width = options.indent }, function ()
    SILE.typesetter:typeset(SILE.formatCounter(SILE.scratch.counters.footnote) .. ".")
  end)
end)
```
This works fine to build an A5 layout of my document, but crashes SILE when building an A4 layout. I suspect t his difference has to do with the width of the insertion frame and some point where it's doing math in one version but not the other, so I'll confine the test to the failing A4 version. The value `options.indent` is a string `14pt`. I've tried hard coding that just to be sure.

Running that code gives this error:

> Error detected:
> /home/caleb/projects/sile//core/length.lua:59: attempt to perform arithmetic on a nil value (local 'other')

If I debug this a little it turns out [the `__sub()` method from length.lua](https://github.com/simoncozens/sile/blob/master/core/length.lua#L48) is getting called with an `other` argument of `nil`. This would lead me to believe that the length calculation in my `pushGlue()` is broken. It's not. That function by itself works fine if I don't  then procede to the `\rebox`. I can comment out **either** the `pushGlue` **or** the `\rebox` and the remaining function will work exactly like it's supposed to with all the correct options. It's only when these two are paired like this that the error is thrown.

If I change [L52](https://github.com/simoncozens/sile/blob/master/core/length.lua#L52) to `other = SILE.toAbsoluteMeasurement(other or 0)` this whole problem goes away, but for the life of me I can't figure out where the `nil` value is coming from because it's not actually my math in this function that's triggering this.

**Comment by pkulchenko** [comment]
FWIW, __sub has `other = SILE.toAbsoluteMeasurement(other or 0)`, but not __add. Is it missing there?

If you are still chasing this bug, I'd check the stack trace from the point where the error is triggered...

**Comment by Omikhleia** [comment]
This issue/question mentions lots of now deprecated methods (`SILE.toAbsoluteMeasurement()`, `SILE.length.parse()` etc.) and doesn't provided a MWE for trying to reproduce and investigate, nor provides a stack trace (as was mentioned in the previous comment - still it would require knowing in which version of SILE this was initially observed, an information not reported either).

Since it's 6+ year old, I'd suggest closing it -- If it is still here (in a different form now), a new and more modern bug report is still possible.

**Comment by alerque** [comment]
This bug was fixed as a side-effect of the overhaul in f4a64e7df.
